click/_compat.py

```diff
@@ -258,6 +258,18 @@ def filename_to_ui(value):
         if isinstance(value, bytes):
             value = value.decode(get_filesystem_encoding(), 'replace')
         return value
+
+    def which(cmd):
+        with open(os.devnull, 'w') as devnull:
+            p = subprocess.Popen(
+                ['which', cmd],
+                stdout=subprocess.PIPE,
+                stderr=devnull,
```

Comment:
Is this required? If there's error output, presumably `wait` doesn't return 0.

Reply:
The `stderr=devnull` was included so as to mimic the `>/dev/null 2>&1` in the current code.  The `which` implementation on my computer doesn't ever seem to write to stderr, but if there's a version that does, we presumably want to supress that so that click users don't see a bunch of "command not found" or whatever messages whenever they call `edit()`.